doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,237 +1257,211 @@ Managing the local context
+      :undocumented:
 
-   .. tacv:: eset (@ident {* @binder } := @term) {? in @goal_occurrences }
-             eset @term {? in @goal_occurrences }
+   .. tacn:: eset @alias_definition {? @occurrences }
+             eset @one_term {? @as_name } {? @occurrences }
       :name: eset; _
 
-      While the different variants of :tacn:`set` expect that no existential
-      variables are generated by the tactic, :tacn:`eset` removes this
-      constraint. In practice, this is relevant only when :tacn:`eset` is
+      Similar to :tacn:`set`, but generates an existential variable when
+      a variable can't be instantiated instead of failing.
+      In practice, this is relevant only when :tacn:`eset` is
       used as a synonym of :tacn:`epose`, i.e. when the :token:`term` does
       not occur in the goal.
 
-.. tacn:: remember @term as @ident__1 {? eqn:@naming_intropattern }
-   :name: remember
+.. tacn:: remember @one_term {? @as_name } {? eqn : @naming_intropattern } {? in @goal_occurrences }
 
-   This behaves as :n:`set (@ident := @term) in *`, using a logical
-   (Leibniz’s) equality instead of a local definition.
+   Equivalent to :n:`set (@ident := @term) in *` but uses :term:`Leibniz equality`
+   instead of a :term:`local definition <context-local definition>`.
    Use :n:`@naming_intropattern` to name or split up the new equation.
 
-   .. tacv:: remember @term as @ident__1 {? eqn:@naming_intropattern } in @goal_occurrences
+   .. tacn:: eremember @one_term {? @as_name } {? eqn : @naming_intropattern } {? in @goal_occurrences }
 
-      This is a more general form of :tacn:`remember` that remembers the
-      occurrences of :token:`term` specified by an occurrence set.
+      Similar to :tacn:`remember`, but generates an existential variable when
+      a variable can't be intantiated rather than causing the tactic to fail.
 
-   .. tacv:: eremember @term as @ident__1 {? eqn:@naming_intropattern } {? in @goal_occurrences }
-      :name: eremember
+.. tacn:: pose @one_term {? @as_name }
+          pose @alias_definition
+   :name: pose; _
 
-      While the different variants of :tacn:`remember` expect that no
-      existential variables are generated by the tactic, :tacn:`eremember`
-      removes this constraint.
+   Adds a :term:`local definition <context-local definition>` to the current context
+   without performing any
+   replacement in the conclusion or hypotheses.  In the first form,
+   the binding :n:`( @ident … := @term )` adds :n:`@term` with the name :n:`@ident`.
+   The second form, :n:`@one_term` with the name either given in :n:`@as_name` or
+   chosen by Coq.  It is equivalent to :n:`set (@ident := @term) in |-`.
 
-.. tacn:: pose (@ident := @term)
-   :name: pose
+   :n:`alias_definition`
+     This form is shorthand for creating local function definitions.
+     :n:`pose ( @ident := fun {* @simple_binder } := @term )` is equivalent to
+     :n:`pose ( @ident := fun {* @simple_binder } => @term)`
```

Comment:
I suggest similar changes in the order of the forms and the description to what you did for `set`.

The note on the equivalent form with `set` makes me realize that we didn't describe the `@occurrences` clause in `set`. Yet, it is very important, since it allows restricting where the term will be replaced by the new variable. FTR, the old doc said "This notation allows specifying which occurrences of :token:`term` have to be substituted in the context.".

Reply:
I changed the order, but I think it's better to point to `set` rather than repeat similar wording. i.e. this is the entire description:

```
   Similar to :n:`set`.  Adds a :term:`local definition <context-local definition>`
   to the context but without doing any replacement.
```